native/packages/react-native-bpk-component-progress/src/BkpProgress.js

```diff
@@ -0,0 +1,159 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2018 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* @flow */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import {
+  StyleSheet,
+  View,
+  Animated,
+  Easing,
+  Platform,
+  ViewPropTypes,
+} from 'react-native';
+import {
+  animmationDurationBase,
+  spacingMd,
+  colorGray100,
+  colorGreen500,
+  colorGreen50,
+  borderRadiusPill,
+  spacingXxl,
+} from 'bpk-tokens/tokens/base.react.native';
+
+const styles = StyleSheet.create({
+  container: {
+    height: spacingMd,
+    overflow: 'hidden',
+  },
+  fill: {
+    height: spacingMd,
+  },
+  defaultTrackStyle: {
+    backgroundColor: Platform.select({
+      ios: () => colorGray100,
+      android: () => colorGreen50,
+    })(),
+    borderRadius: borderRadiusPill,
+    width: spacingXxl * 3,
+  },
+  defaultFillStyle: {
+    backgroundColor: colorGreen500,
+    borderRadius: borderRadiusPill,
+  },
+  barTrackStyle: {
+    backgroundColor: Platform.select({
+      ios: () => colorGray100,
+      android: () => colorGreen50,
+    })(),
+    width: '100%',
+  },
+  barFillStyle: {
+    backgroundColor: colorGreen500,
+  },
+});
+
+type Props = {
+  max: number,
+  min: number,
+  value: number,
+  type: 'default' | 'bar',
```

Comment:
We've been using consts for this sort of thing to avoid repetition, here's an example from the button component: https://github.com/Skyscanner/backpack/blob/master/native/packages/react-native-bpk-component-button/src/common-types.js#L35.

So in this case, you could do:

```
const BAR_TYPES = {
  default: 'default',
  trailing: 'trailing',
};

type Props = {
  ...
  type: $Keys<typeof BAR_TYPES>,
};

static propTypes = {
  ...
  type: PropTypes.oneOf(Object.keys(BAR_TYPES)),
};

static defaultProps = {
  ...
  type: BAR_TYPES.default,
};
```